spring-cloud-gcp-starters/spring-cloud-gcp-starter-config/src/main/java/org/springframework/cloud/gcp/config/GoogleConfigPropertySourceLocator.java

```diff
@@ -0,0 +1,148 @@
+import org.springframework.cloud.config.client.ConfigClientProperties;
+import org.springframework.cloud.config.client.ConfigServicePropertySourceLocator;
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.util.Assert;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Custom {@link PropertySourceLocator} for Google Cloud Runtime Configurator API.
+ *
+ * @author Jisha Abubaker
+ */
+public class GoogleConfigPropertySourceLocator implements PropertySourceLocator {
+
+	private static final String RUNTIMECONFIG_API_ROOT = "https://runtimeconfig.googleapis.com/v1beta1/";
+
+	private static final String ALL_VARIABLES_PATH =
+			"projects/{project}/configs/{name}_{profile}/variables?returnValues=true";
+
+	private static final String PROPERTY_SOURCE_NAME = "spring-cloud-gcp";
+
+	private static final String AUTHORIZATION_HEADER = "Authorization";
+
+	private static final String BEARER_TOKEN_PREFIX = "Bearer ";
+
+	private String projectId;
+
+	private Credentials credentials;
+
+	private ConfigClientProperties configClientProperties;
+
+	private int timeout;
+
+	public GoogleConfigPropertySourceLocator(GcpProjectIdProvider projectIdProvider,
+			CredentialsProvider credentialsProvider,
+			ConfigClientProperties configClientProperties,
+			GcpConfigProperties gcpConfigProperties) throws Exception {
+		Assert.notNull(credentialsProvider, "Credentials provider cannot be null");
+		Assert.notNull(projectIdProvider, "Project ID provider cannot be null");
+		this.credentials = credentialsProvider.getCredentials();
+		this.projectId = projectIdProvider.getProjectId();
+		Assert.notNull(this.credentials, "Credentials must not be null");
+		Assert.notNull(this.projectId, "Project ID must not be null");
+
+		this.timeout = 60000; // 1 minute in milliseconds
+		if (gcpConfigProperties != null) {
+			this.timeout = gcpConfigProperties.getTimeout();
+		}
+		this.configClientProperties = configClientProperties;
+		Assert.notNull(this.configClientProperties, "Client Configuration must not be null");
+		Assert.notNull(this.timeout, "Runtime Configurator API timeout must not be null");
+	}
+
+	private RestTemplate getSecureRestTemplate() throws Exception {
+		SimpleClientHttpRequestFactory requestFactory = new SimpleClientHttpRequestFactory();
+		requestFactory.setReadTimeout(this.timeout);
+		RestTemplate template = new RestTemplate(requestFactory);
+		Map<String, String> headers = new HashMap<>();
+		Map<String, List<String>> requestMetadata = this.credentials.getRequestMetadata();
```

Comment:
Can we log somewhere in this class (maybe here? or in the ctor?) what user we're authenticating as? Kind of like we do [here](https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfiguration.java#L99).

I'm debugging an issue where the application default credentials are giving me 403 Forbiddens and knowing who the users are authenticating as here would be really useful to debug issues.